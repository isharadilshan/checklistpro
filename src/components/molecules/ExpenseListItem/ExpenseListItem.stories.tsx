import React from 'react';
import {NativeBaseProvider} from 'native-base';
import {ComponentMeta, ComponentStory} from '@storybook/react';

import ExpenseListItem from './';

export default {
  title: 'components/molecules/ExpenseListItem',
  component: ExpenseListItem,
} as ComponentMeta<typeof ExpenseListItem>;

export const Basic: ComponentStory<typeof ExpenseListItem> = () => {
  const args = {
    id: 'EXPENSE_ID',
    title: 'Expense Title',
    description: 'Expense Description',
    amount: 402,
    category: 'MEDICAL',
    createdDate: 1680101338199,
    updatedDate: 1680101338199,
    latitude: 12.3455,
    longitude: 12.3455,
    hideButtons: false,
    onPressEdit: () => {},
    onPressDelete: () => {},
  };
  return (
    <NativeBaseProvider>
      <ExpenseListItem {...args} />
    </NativeBaseProvider>
  );
};
