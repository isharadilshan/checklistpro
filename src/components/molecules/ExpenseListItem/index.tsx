import React from 'react';
import {Box, Badge, HStack, View, Text} from 'native-base';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';

type ExpenseListItemProps = {
  id: string;
  title: string;
  description: string;
  amount: number;
  category: string;
  createdDate: number;
  hideButtons?: boolean;
  onPressEdit: (id: string) => void;
  onPressDelete: (id: string) => void;
};

const ExpenseListItem: React.FC<ExpenseListItemProps> = ({
  id,
  title,
  description,
  amount,
  category,
  createdDate,
  hideButtons,
  onPressEdit,
  onPressDelete,
}) => {
  return (
    <Box
      height={150}
      rounded="sm"
      overflow="hidden"
      borderColor="coolGray.200"
      borderWidth="1"
      _dark={{
        borderColor: 'coolGray.600',
        backgroundColor: '#1f2937',
      }}
      _light={{
        backgroundColor: '#1f2937',
      }}
      p={2}
      my={2}
    >
      <HStack height={'100%'}>
        <Box w={'75%'}>
          <View style={{flex: 1, justifyContent: 'space-between'}}>
            <View>
              <Text color={'blueGray.400'} fontSize="xl">
                {title}
              </Text>
              <Text color={'blueGray.400'}>{description}</Text>
            </View>
            <Badge width={150} variant={'outline'} colorScheme="success">
              {category}
            </Badge>
          </View>
        </Box>
        {!hideButtons && (
          <Box w={'25%'}>
            <View
              style={{
                flex: 1,
                alignItems: 'flex-end',
                justifyContent: 'space-between',
              }}
            >
              <Icon
                name="square-edit-outline"
                size={30}
                color="#94a3b8"
                onPress={() => onPressEdit(id)}
              />
              <Icon
                name="delete"
                size={30}
                color="#94a3b8"
                onPress={() => onPressDelete(id)}
              />
            </View>
          </Box>
        )}
      </HStack>
    </Box>
  );
};

export default ExpenseListItem;
