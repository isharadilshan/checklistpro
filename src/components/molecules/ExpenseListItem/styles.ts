import {ViewStyle, StyleSheet} from 'react-native';

interface Style {
  cardWrapper: ViewStyle;
  textWrapper: ViewStyle;
  buttonWrapper: ViewStyle;
  iconStyle: ViewStyle;
}

export default () => {
  return StyleSheet.create<Style>({
    cardWrapper: {
      height: 150,
      borderWidth: 1,
      borderRadius: 4,
      padding: 10,
      marginVertical: 10,
      overflow: 'hidden',
      borderColor: '#e5e7eb',
      backgroundColor: '#1f2937',
    },
    textWrapper: {
      flex: 1,
      justifyContent: 'space-between',
    },
    buttonWrapper: {
      flex: 1,
      alignItems: 'flex-end',
      justifyContent: 'space-between',
    },
    iconStyle: {
      color: '#94a3b8',
    },
  });
};
