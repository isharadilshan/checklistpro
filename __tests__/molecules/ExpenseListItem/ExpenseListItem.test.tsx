import React from 'react';
import {describe, expect} from '@jest/globals';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import ExpenseListItem from '../../../src/components/molecules/ExpenseListItem';

configure({adapter: new Adapter()});

const props = {
  id: 'EXPENSE_ID',
  title: 'Expense Title',
  description: 'Expense Description',
  amount: 402,
  category: 'MEDICAL',
  createdDate: 1680101338199,
  updatedDate: 1680101338199,
  latitude: 12.3455,
  longitude: 12.3455,
  hideButtons: false,
  onPressEdit: jest.fn(),
  onPressDelete: jest.fn(),
};

describe('when rendering', () => {
  const wrapper = shallow(<ExpenseListItem {...props} />);

  it('should render a component matching the snapshot', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
    expect(wrapper).toHaveLength(1);
  });
});
