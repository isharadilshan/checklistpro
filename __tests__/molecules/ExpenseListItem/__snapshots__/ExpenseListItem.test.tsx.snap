// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rendering should render a component matching the snapshot 1`] = `
<Memo(ForwardRef(Box))
  style={
    {
      "backgroundColor": "#1f2937",
      "borderColor": "#e5e7eb",
      "borderRadius": 4,
      "borderWidth": 1,
      "height": 150,
      "marginVertical": 10,
      "overflow": "hidden",
      "padding": 10,
    }
  }
>
  <Memo(ForwardRef(HStack))
    height="100%"
  >
    <Memo(ForwardRef(Box))
      w="50%"
    >
      <ForwardRef
        style={
          {
            "flex": 1,
            "justifyContent": "space-between",
          }
        }
      >
        <ForwardRef>
          <Memo(ForwardRef(Text))
            color="blueGray.400"
            fontSize="xl"
          >
            Expense Title
          </Memo(ForwardRef(Text))>
          <Memo(ForwardRef(Text))
            color="blueGray.400"
          >
            Expense Description
          </Memo(ForwardRef(Text))>
          <Memo(ForwardRef(Text))
            color="blueGray.400"
          >
            Wednesday, March 29, 2023 8:18 PM
          </Memo(ForwardRef(Text))>
        </ForwardRef>
        <Memo(ForwardRef(Badge))
          colorScheme="success"
          variant="outline"
          width={150}
        >
          MEDICAL
        </Memo(ForwardRef(Badge))>
      </ForwardRef>
    </Memo(ForwardRef(Box))>
    <Memo(ForwardRef(Box))
      w="25%"
    >
      <Memo(ForwardRef(Text))
        color="blueGray.400"
        fontSize="2xl"
      >
        402
      </Memo(ForwardRef(Text))>
    </Memo(ForwardRef(Box))>
    <Memo(ForwardRef(Box))
      w="25%"
    >
      <ForwardRef
        style={
          {
            "alignItems": "flex-end",
            "flex": 1,
            "justifyContent": "space-between",
          }
        }
      >
        <Icon
          allowFontScaling={false}
          name="square-edit-outline"
          onPress={[Function]}
          size={30}
          style={
            {
              "color": "#94a3b8",
            }
          }
        />
        <Icon
          allowFontScaling={false}
          name="delete"
          onPress={[Function]}
          size={30}
          style={
            {
              "color": "#94a3b8",
            }
          }
        />
      </ForwardRef>
    </Memo(ForwardRef(Box))>
  </Memo(ForwardRef(HStack))>
</Memo(ForwardRef(Box))>
`;
